src/components/TodoListScreen.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import {
+  Button,
+  StyleSheet,
+  View,
+  Text,
+  TouchableOpacity,
+} from 'react-native';
+
+class TodoListScreen extends Component {
+  render(){
+    const { navigation, allTodos} = this.props;
+    const todosList = allTodos.map((todo, index) => 
+      <Text key={index} style={styles.todoItem}>{todo}</Text>);
+    return (
+      <View style={styles.container}>
+          <Text style={{ fontSize: 30, height: "10%" }}>To Do</Text>
+          { todosList }
```

Comment:
Podes fazer o map diretamente no render
```suggestion
          {allTodos.map((todo, index) => (
             <Text key={index} style={styles.todoItem}>{todo}</Text>))
          }
```